packages/iov-core/src/signingservercore.ts

```diff
@@ -9,29 +9,57 @@ import { UserProfile, WalletId } from "@iov/keycontrol";
 
 import { MultiChainSigner } from "./multichainsigner";
 
+export interface GetIdentitiesAuthorization {
+  /**
+   * Called by the signing server and lets user decide which of the
+   * available identities should be revealed to the application.
+   */
+  (reason: string, matchingIdentities: ReadonlyArray<PublicIdentity>): Promise<ReadonlyArray<PublicIdentity>>;
+}
+
+export interface SignAndPostAuthorization {
+  /**
+   * Called by the signing server and lets user decide if they want to
+   * authorize a sign and post request
+   */
+  (reason: string, transaction: UnsignedTransaction): Promise<boolean>;
+}
+
 export class SigningServerCore {
   private readonly signer: MultiChainSigner;
   private readonly profile: UserProfile;
+  private readonly authorizeGetIdentities: GetIdentitiesAuthorization;
+  private readonly authorizeSignAndPost: SignAndPostAuthorization;
 
-  constructor(profile: UserProfile, signer: MultiChainSigner) {
+  constructor(
+    profile: UserProfile,
+    signer: MultiChainSigner,
+    authorizeGetIdentities: GetIdentitiesAuthorization,
+    authorizeSignAndPost: SignAndPostAuthorization,
+  ) {
     this.signer = signer;
     this.profile = profile;
+    this.authorizeGetIdentities = authorizeGetIdentities;
+    this.authorizeSignAndPost = authorizeSignAndPost;
   }
 
   public async getIdentities(
-    _: string,
+    reason: string,
     chainIds: ReadonlyArray<ChainId>,
   ): Promise<ReadonlyArray<PublicIdentity>> {
     const matchingIdentities = this.allIdentities().filter(identity => {
       return chainIds.some(chainId => identity.chainId === chainId);
     });
 
-    // TODO: ask user for permission and allow selection of identities
+    const authorizedIdentities = this.authorizeGetIdentities(reason, matchingIdentities);
 
-    return matchingIdentities;
+    return authorizedIdentities;
```

Comment:
Why the const and not just return directly above?

Reply:
No specific reason, both ways are fine. Having a variable name allowed me to express the meaning of the callback's return value